JuliaInterface/src/JuliaInterface.c

```diff
@@ -207,6 +207,107 @@ Obj NewJuliaFunc(jl_function_t* function, int autoConvert)
     return func;
 }
 
+/*
+ * C function pointer calls
+ */
+
+typedef Obj (*func_ptr_0_args)();
+Obj DoCallJuliaCFunc0Arg( Obj func )
+{
+    func_ptr_0_args function = (func_ptr_0_args)jl_unbox_voidpointer((jl_value_t*)FEXS_FUNC(func));
+    return function();
+}
+
+typedef Obj (*func_ptr_1_args)(Obj);
+Obj DoCallJuliaCFunc1Arg( Obj func, Obj arg1 )
+{
+    func_ptr_1_args function = (func_ptr_1_args)jl_unbox_voidpointer((jl_value_t*)FEXS_FUNC(func));
+    return function(arg1);
+}
+
+typedef Obj (*func_ptr_2_args)(Obj,Obj);
+Obj DoCallJuliaCFunc2Arg( Obj func, Obj arg1, Obj arg2 )
+{
+    func_ptr_2_args function = (func_ptr_2_args)jl_unbox_voidpointer((jl_value_t*)FEXS_FUNC(func));
+    return function(arg1,arg2);
+}
+
+typedef Obj (*func_ptr_3_args)(Obj,Obj,Obj);
+Obj DoCallJuliaCFunc3Arg( Obj func, Obj arg1, Obj arg2, Obj arg3 )
+{
+    func_ptr_3_args function = (func_ptr_3_args)jl_unbox_voidpointer((jl_value_t*)FEXS_FUNC(func));
+    return function(arg1,arg2,arg3);
+}
+
+typedef Obj (*func_ptr_4_args)(Obj,Obj,Obj,Obj);
+Obj DoCallJuliaCFunc4Arg( Obj func, Obj arg1, Obj arg2, Obj arg3, Obj arg4 )
+{
+    func_ptr_4_args function = (func_ptr_4_args)jl_unbox_voidpointer((jl_value_t*)FEXS_FUNC(func));
+    return function(arg1,arg2,arg3,arg4);
+}
+
+typedef Obj (*func_ptr_5_args)(Obj,Obj,Obj,Obj,Obj);
+Obj DoCallJuliaCFunc5Arg( Obj func, Obj arg1, Obj arg2, Obj arg3, Obj arg4, Obj arg5 )
+{
+    func_ptr_5_args function = (func_ptr_5_args)jl_unbox_voidpointer((jl_value_t*)FEXS_FUNC(func));
+    return function(arg1,arg2,arg3,arg4,arg5);
+}
+
+typedef Obj (*func_ptr_6_args)(Obj,Obj,Obj,Obj,Obj,Obj);
+Obj DoCallJuliaCFunc6Arg( Obj func, Obj arg1, Obj arg2, Obj arg3, Obj arg4, Obj arg5, Obj arg6 )
+{
+    func_ptr_6_args function = (func_ptr_6_args)jl_unbox_voidpointer((jl_value_t*)FEXS_FUNC(func));
+    return function(arg1,arg2,arg3,arg4,arg5,arg6);
+}
+
+Obj NewJuliaCFunc(void* function, int nr_args, char* args)
+{
+
+    Obj func = NewFunctionC("", nr_args, args, 0);
+
+    switch(nr_args){
+        case 0:
```

Comment:
Off-topic, but: it'd be great if there was a `.clang-format` file in this repo. Then the code in this file could be formatted uniformly. I normally try to adapt to whatever code formatting conventions are used by a project, but here, I really can't tell what those are, resp. find them so different from any I used before that I have a hard time following them. In either case, a `.clang-format` would simplify writing patches for me a lot.